bundle-workflow/src/test_workflow/test_recorder.py

```diff
@@ -3,34 +3,133 @@
 # The OpenSearch Contributors require contributions made to
 # this file be licensed under the Apache-2.0 license or a
 # compatible open source license.
-
 import logging
+import os
+import shutil
+import tempfile
+
+import yaml
 
 
 class TestRecorder:
```

Comment:
Make this class a singleton. With the design pattern we have, we'd like to have just one instance of TestRecorder for a given testsuite/jenkins job. 
This also prevents other instantiations, avoids passing the test recorder instance in multiple places within the code base.

```suggestion
class TestRecorder:
   __instance = None
   @staticmethod 
   def getInstance(test_run_id, test_type, location=None):
      """ Static access method. """
      if TestRecorder.__instance == None:
         TestRecorder(test_run_id, test_type, location)
      return TestRecorder.__instance
```

Python by default doesn't have private constructors, if you'd like to block instantiation outside of the class, you could raise an error in `__init__` and use a different private method which could be called by `getInstance(...)`

Reply:
Also rename getInstance to get_instance